src/components/Modal/Modal.tsx

```diff
@@ -1,34 +1,27 @@
 import { Dialog as BaseDialog } from '@kobalte/core';
-import { ComponentProps, ParentComponent, splitProps } from 'solid-js';
+import type { Component, ComponentProps } from 'solid-js';
 
-import { buttonRecipe, ButtonVariants } from '@cubeartisan/cubeartisan/components/Button';
 import * as styles from '@cubeartisan/cubeartisan/components/Modal/Modal.css';
 
-const ModalTrigger: ParentComponent<{ recipe?: ButtonVariants } & ComponentProps<typeof BaseDialog.Trigger>> = (
-  props,
-) => {
-  const [local, others] = splitProps(props, ['recipe']);
+const ModalTrigger = BaseDialog.Trigger;
 
-  return <BaseDialog.Trigger class={buttonRecipe(local.recipe)} {...others} />;
-};
-
-const ModalCloseButton: ParentComponent<{ recipe?: ButtonVariants } & ComponentProps<typeof BaseDialog.CloseButton>> = (
-  props,
-) => {
-  const [local, others] = splitProps(props, ['recipe']);
-
-  return <BaseDialog.CloseButton class={buttonRecipe(local.recipe)} {...others} />;
-};
+const ModalCloseButton = BaseDialog.CloseButton;
 
 const ModalPortal = BaseDialog.Portal;
 
-const ModalOverlay: typeof BaseDialog.Overlay = (props) => <BaseDialog.Overlay class={styles.overlay} {...props} />;
+const ModalOverlay: Component<ComponentProps<typeof BaseDialog.Overlay>> = (props) => (
+  <BaseDialog.Overlay class={styles.overlay} {...props} />
+);
 
-const ModalContent: typeof BaseDialog.Content = (props) => <BaseDialog.Content class={styles.content} {...props} />;
+const ModalContent: Component<ComponentProps<typeof BaseDialog.Content>> = (props) => (
```

Comment:
Same as above also how does this handle if props contains a class prop?

Reply:
you are right that I need to have a splitprops here. I will fix that

Reply:
fixed